examples/microwave-oven-app/microwave-oven-common/include/microwave-oven-device.h

```diff
@@ -69,28 +61,60 @@ class ExampleMicrowaveOvenDevice : public MicrowaveOvenControl::Delegate,
     /**
      * handle command for microwave oven control: set cooking parameters
      */
-    Protocols::InteractionModel::Status HandleSetCookingParametersCallback(uint8_t cookMode, uint32_t cookTime,
-                                                                           uint8_t powerSetting) override;
+    Protocols::InteractionModel::Status HandleSetCookingParametersCallback(uint8_t cookMode, uint32_t cookTimeSec,
+                                                                           bool startAfterSetting,
+                                                                           Optional<uint8_t> powerSettingNum,
+                                                                           Optional<uint8_t> wattSettingIndex) override;
 
     /**
      * handle command for microwave oven control: add more time
      */
-    Protocols::InteractionModel::Status HandleModifyCookTimeCallback(uint32_t finalCookTime) override;
+    Protocols::InteractionModel::Status HandleModifyCookTimeSecondsCallback(uint32_t finalcookTimeSec) override;
+
+    /**
+     *   Get the watt setting from the supported watts array.
+     *   @param index The index of the watt setting to be returned. It is assumed that watt setting are indexable from 0 and with no
+     * gaps.
+     *   @param wattSetting A reference to receive the watt setting on success.
+     *   @return Returns a CHIP_NO_ERROR if there was no error and the label was returned successfully.
+     *   CHIP_ERROR_NOT_FOUND if the index in beyond the list of available labels.
+     */
+    CHIP_ERROR GetWattSettingByIndex(uint8_t index, uint16_t & wattSetting) override;
+
+    /**
+     * Get the value of power setting number.
+     */
+    uint8_t GetPowerSettingNum() const override { return mPowerSettingNum; }
+
+    /**
+     * Get the value of min power number.
+     */
+    uint8_t GetMinPowerNum() const override { return kMinPowerNum; }
 
     /**
-     * Get the value of MinPower.
+     * Get the value of max power number.
      */
-    uint8_t GetMinPower() const override { return kDefaultMinPower; }
+    uint8_t GetMaxPowerNum() const override { return kMaxPowerNum; }
 
     /**
-     * Get the value of MaxPower.
+     * Get the value of power step number.
      */
-    uint8_t GetMaxPower() const override { return kDefaultMaxPower; }
+    uint8_t GetPowerStepNum() const override { return kPowerStepNum; }
 
     /**
-     * Get the value of PowerStep.
+     * Get the value of max cook time in seconds.
      */
-    uint8_t GetPowerStep() const override { return kDefaultPowerStep; }
+    uint32_t GetMaxCookTimeSec() const override { return kMaxCookTimeSec; }
+
+    /**
+     * Get the value of selected watt index.
+     */
+    uint8_t GetCurrentWattIndex() const override { return mSelectedWattIndex; };
+
+    /**
+     * Get the value of watt rating.
+     */
+    uint16_t GetCurrentWattRating() const override { return mWattRating; };
```

Comment:
The watt rating is fixed.  Recommend changing this name to `GetWattRating`.

```suggestion
    uint16_t GetWattRating() const override { return mWattRating; };
```